depthmapX/mainwindowpluginfactory.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2020, Petros Koutsolampros
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Original found here: https://stackoverflow.com/a/582456
+
+#pragma once
+
+#include "depthmapX/mainwindowplugin.h"
+
+#include <iostream>
+
+template <typename T> MainWindowPlugin *createT() { return new T; }
+
+struct MainWindowPluginFactory {
+    typedef std::map<std::string, MainWindowPlugin *(*)()> PluginMap;
+
+    static MainWindowPlugin *createInstance(std::string const &s) {
```

Comment:
That method is a misnomer: it does not create instances, it just returns pre-existing instances